### PR TITLE
Magmite Vacpack, A Ceiros Drop

### PR DESCRIPTION
**Pull request by SkippZzMods**
What are the specifics of what is being added?
The Magmite Vacpack is being added, which is a summon weapon that launches sticky Magmites that when stuck on enemies, causes them to take stacking summon tag damage and DoT.

Is there an associated suggestion or dev server #code-needed post? How well does this PR follow it?
This was from the Trello. I tried to make it as true to the Trello as possible

Is there anything left to be done in this addition?
There could maybe be some better sound effects.

### CODE REVIEW

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+            return new Vector2(-10f, 0f);
+        }
+    }
+
+    //literally just for tank drawing lol
+    internal class MagmiteVacpackPlayer : ModPlayer
+    {
+        public override void ModifyDrawInfo(ref PlayerDrawSet drawInfo)
+        {
+            if (drawInfo.shadow != 0f)
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:61]: single line conditionals should be inlined

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+            }
+        }
+    }
+
+    internal class MagmiteVacpackHoldout : ModProjectile
+    {
+        private int time;
+
+        public ref float MaxFramesTillShoot => ref Projectile.ai[1];
+
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:92]: probably shouldn't have this extra linebreak here

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+                Projectile.Kill();
+
+            if (ShootDelay >= MaxFramesTillShoot)
+            {
+                Item heldItem = owner.HeldItem;
+                int damage = Projectile.damage;
+                float shootSpeed = heldItem.shootSpeed;
+                float knockBack = owner.GetWeaponKnockback(heldItem, heldItem.knockBack);
+                Vector2 shootVelocity = Utils.SafeNormalize(Projectile.velocity, Vector2.UnitY) * shootSpeed;
+                if (Main.myPlayer == Projectile.owner)
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:144]: single line conditionals should be inlined

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+            }
+        }
+
+        public override bool OnTileCollide(Vector2 oldVelocity)
+        {
+            MaxBounces--;
+            if (MaxBounces <= 0)
+                Projectile.Kill();
+
+            if (Math.Abs(Projectile.velocity.X - oldVelocity.X) > float.Epsilon)
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:326]: Single line conditionals should be inlined

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+                if (damage < 1)
+                    damage = 1;
+            }
+        }
+
+        public override void ModifyHitByProjectile(NPC npc, Projectile projectile, ref int damage, ref float knockback, ref bool crit, ref int hitDirection)
+        {
+            Player player = Main.player[projectile.owner];
+
+            if (projectile.owner == MagmiteOwner && projectile.friendly && (projectile.minion || projectile.DamageType == DamageClass.Summon || ProjectileID.Sets.MinionShot[projectile.type] == true) && npc.whoAmI == player.MinionAttackTargetNPC && MagmiteAmount > 0 && player.HasMinionAttackTargetNPC)
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:442]: This condition is a bit big and hard to read, is there a way to simplify it?

**Review by ScalarVector1 — CHANGES_REQUESTED** [review]
A couple of minor style changes needed, but otherwise looks fine.

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 6]
```diff
@@ -439,7 +429,9 @@ public override void ModifyHitByProjectile(NPC npc, Projectile projectile, ref i
         {
             Player player = Main.player[projectile.owner];
 
-            if (projectile.owner == MagmiteOwner && projectile.friendly && (projectile.minion || projectile.DamageType == DamageClass.Summon || ProjectileID.Sets.MinionShot[projectile.type] == true) && npc.whoAmI == player.MinionAttackTargetNPC && MagmiteAmount > 0 && player.HasMinionAttackTargetNPC)
+            bool IsMinion = (projectile.minion || projectile.DamageType == DamageClass.Summon || ProjectileID.Sets.MinionShot[projectile.type] == true);
+
+            if (projectile.owner == MagmiteOwner && projectile.friendly && IsMinion && npc.whoAmI == player.MinionAttackTargetNPC && MagmiteAmount > 0 && player.HasMinionAttackTargetNPC)
             {
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:435]: Single line conditionals should be inlined

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs [hunk 7]
```diff
@@ -439,7 +429,9 @@ public override void ModifyHitByProjectile(NPC npc, Projectile projectile, ref i
         {
             Player player = Main.player[projectile.owner];
 
-            if (projectile.owner == MagmiteOwner && projectile.friendly && (projectile.minion || projectile.DamageType == DamageClass.Summon || ProjectileID.Sets.MinionShot[projectile.type] == true) && npc.whoAmI == player.MinionAttackTargetNPC && MagmiteAmount > 0 && player.HasMinionAttackTargetNPC)
+            bool IsMinion = (projectile.minion || projectile.DamageType == DamageClass.Summon || ProjectileID.Sets.MinionShot[projectile.type] == true);
```
Comment (ScalarVector1) [inline Content/Items/Vitric/Weapons.MagmiteVacpack.cs:432]: This name seems misleading, why would it be a minion if its a minion or a summon projectile? (whips are summon projectiles!)

**Review by ScalarVector1 — COMMENTED** [review]
ModifyHitByProjectile in the globalNPC of the vacpac still seems a bit strange. Its looking like you may be checking some incorrect conditions?

File: Content/Items/Vitric/Weapons.MagmiteVacpack.cs (hunk 7 above)
Comment (SkippZzMods) [reply Content/Items/Vitric/Weapons.MagmiteVacpack.cs:432]: by IsMinion I just meant that its a minion damage type, I should have, and will rename to bool to IsSummon. Can you elaborate on the wrong conditions that Im checking?

Comment (ScalarVector1) [reply Content/Items/Vitric/Weapons.MagmiteVacpack.cs:432]: I was mostly reffering to detecting summoner projectiles in general as being potentially incorrect. If thats intentional than you are fine I believe.

Comment (SkippZzMods) [reply Content/Items/Vitric/Weapons.MagmiteVacpack.cs:432]: Yes I was intending to detect all summoner projectiles in general, I should have named the bool more clearly though, thats my bad. Will be fixed asap

**Pull request merged**